blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorManager.java

```diff
@@ -5,92 +5,87 @@
 import android.content.Intent;
 import android.content.ServiceConnection;
 import android.os.IBinder;
-import android.util.Log;
-import android.widget.Toast;
 
-import com.google.blockly.android.codegen.CodeGenerationRequest.CodeGeneratorCallback;
-import com.google.blockly.model.BlocklySerializerException;
-import com.google.blockly.model.Workspace;
-import com.google.blockly.utils.StringOutputStream;
-
-import java.util.List;
+import java.util.LinkedList;
+import java.util.Queue;
 
 /**
  * Client Side class responsible for connecting to the {@link CodeGeneratorService}.
+ *
+ * A connection to the service is only made the first time code generation is requested.
  */
 public class CodeGeneratorManager {
     private static final String TAG = "CodeGeneratorManager";
 
     private final Context mContext;
-    private final ServiceConnection mCodeGenerationConnection;
+    private final Queue<CodeGenerationRequest> mStoredRequests;
 
+    private ServiceConnection mCodeGenerationConnection;
     private CodeGeneratorService mGeneratorService;
 
     public CodeGeneratorManager(Context context) {
         this.mContext = context;
-
-        this.mCodeGenerationConnection = new ServiceConnection() {
-            @Override
-            public void onServiceConnected(ComponentName className, IBinder binder) {
-                mGeneratorService =
-                    ((CodeGeneratorService.CodeGeneratorBinder) binder).getService();
-            }
-
-            @Override
-            public void onServiceDisconnected(ComponentName arg0) {
-                mGeneratorService = null;
-            }
-        };
+        this.mStoredRequests = new LinkedList<>();
     }
 
+    /**
+     * Resumes the underlying code generation connection.
+     */
     public void onResume() {
         Intent intent = new Intent(mContext, CodeGeneratorService.class);
-        mContext.bindService(intent, mCodeGenerationConnection, Context.BIND_AUTO_CREATE);
+        if (mCodeGenerationConnection != null) {
+            mContext.bindService(intent, mCodeGenerationConnection, Context.BIND_AUTO_CREATE);
```

Comment:
To be clear, it is this bind that could be lazy, and is the point of issue #37. We should wait until the first `requestCodeGeneration()` call, since it is entirely possible Blockly is used without code generation, and a full background WebView is fairly heavyweight. Scratch Blocks is an example of this sort of usage.